test/liquidity-protection/LiquidityProtection.js

```diff
@@ -3802,6 +3426,565 @@ describe('LiquidityProtection', () => {
+             * - total amount of YYY/BNT pool tokens
+             * - total amount of ETH/BNT pool tokens
+             * - total amount of XXX in the XXX/BNT pool
+             * - total amount of BNT in the XXX/BNT pool
+             * - total amount of YYY in the YYY/BNT pool
+             * - total amount of BNT in the YYY/BNT pool
+             * - total amount of ETH in the ETH/BNT pool
+             * - total amount of BNT in the ETH/BNT pool
+             * - per provider amount of XXX in the XXX/BNT pool
+             * - per provider amount of BNT in the XXX/BNT pool
+             * - per provider amount of YYY in the YYY/BNT pool
+             * - per provider amount of BNT in the YYY/BNT pool
+             * - per provider amount of ETH in the ETH/BNT pool
+             * - per provider amount of BNT in the ETH/BNT pool
+             */
+            describe('migration tests', () => {
+                const NUM_OF_POOLS = 3;
+                const NUM_OF_PROVIDERS = 3;
+                const NUM_OF_POSITIONS = 9; // number of positions per provider per pool per reserve
+
+                let providers;
+                let baseTokens;
+
+                const filter = (position, poolToken, reserveToken) =>
+                    position.poolToken === poolToken.address && position.reserveToken === reserveToken.address;
+
+                const getPositions = (positions, poolToken, reserveToken) =>
+                    positions.filter((position) => filter(position, poolToken, reserveToken));
+
+                const getMigrationUnit = (positionIds, positions, poolToken, reserveToken) => ({
+                    poolToken: poolToken.address,
+                    reserveToken: reserveToken.address,
+                    positionIds: positionIds.filter((_, i) => filter(positions[i], poolToken, reserveToken))
+                });
+
+                const poolAvailableSpace = async (poolToken, index, percent) =>
+                    (await liquidityProtection.poolAvailableSpace(poolToken.address))[index].mul(percent).div(100);
+
+                const readState = async (baseToken, provider) => ({
+                    providerGovBalance: await govToken.balanceOf(provider.address),
+                    providerBaseBalance: await baseToken.balanceOf(provider.address),
+                    providerNetworkBalance: await networkToken.balanceOf(provider.address),
+                    providerPoolBalance: await baseToken.poolToken.balanceOf(provider.address),
+                    vaultGovBalance: await govToken.balanceOf(bancorVault.address),
+                    vaultBaseBalance: await baseToken.balanceOf(bancorVault.address),
+                    vaultNetworkBalance: await networkToken.balanceOf(bancorVault.address),
+                    vaultPoolBalance: await baseToken.poolToken.balanceOf(bancorVault.address),
+                    walletGovBalance: await govToken.balanceOf(liquidityProtectionWallet.address),
```

Comment:
we can call it protectionGovBalance etc. instead of walletGovBalance which is very generic

Reply:
Done